src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -406,17 +409,21 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
         ssMS << RangeToString(", ", "gridtools::define_caches(", "),")(
             multiStage.getCaches(),
             [&](const std::pair<int, Cache>& AccessIDCachePair) -> std::string {
+              auto const& cache = AccessIDCachePair.second;
+              DAWN_ASSERT(cache.getInterval().is_initialized() ||
```

Comment:
can we assert? should we not just ignore it if it was already initialized?

Reply:
If the cache is local, then we dont need any interval, since we dont synchronize with main memory. 
Otherwise we need an interval, that means that the boost::optional<> has to be initialized (i.e. not null)

Reply:
yeah, but if we find a local cache, we should not assert, we should just not anything, right?

Reply:
yes, that is why we assert if is_initialized or policy is local (line below)